File: backend/src/routes/meta.ts
import type { FastifyInstance } from "fastify";

import { LOCAL_CONSOLE_QUERY_API_VERSION } from "../../../shared/src/enums.js";

export interface LocalConsoleCapabilities {
  tokenUsageEnabled: boolean;
  gatewayAuthLogsEnabled: boolean;
}

export function registerMetaRoutes(
  app: FastifyInstance,
  capabilities: LocalConsoleCapabilities = {
    tokenUsageEnabled: false,
    gatewayAuthLogsEnabled: false,
  },
): void {
  app.get("/meta/capabilities", async () => ({
    tokenUsageEnabled: capabilities.tokenUsageEnabled,
    gatewayAuthLogsEnabled: capabilities.gatewayAuthLogsEnabled,
    queryApiVersion: LOCAL_CONSOLE_QUERY_API_VERSION,
  }));
}

File: backend-go/internal/routes/meta.go
package routes

import (
	"github.com/gin-gonic/gin"
)

const queryAPIVersion = "v1"

// MetaCapabilities mirrors LocalConsoleCapabilities.
type MetaCapabilities struct {
	TokenUsageEnabled      bool
	GatewayAuthLogsEnabled bool
}

// RegisterMeta mirrors registerMetaRoutes.
func RegisterMeta(router gin.IRoutes, cap MetaCapabilities) {
	router.GET("/meta/capabilities", func(c *gin.Context) {
		c.JSON(200, map[string]any{
			"tokenUsageEnabled":      cap.TokenUsageEnabled,
			"gatewayAuthLogsEnabled": cap.GatewayAuthLogsEnabled,
			"queryApiVersion":        queryAPIVersion,
		})
	})
}
